internal/transport/grpchttp2/framer.go

```diff
@@ -0,0 +1,317 @@
+type PingFrame struct {
+	hdr  *FrameHeader
+	free func()
+	Data []byte
+}
+
+func (f *PingFrame) Header() *FrameHeader {
+	return f.hdr
+}
+
+func (f *PingFrame) Free() {
+	if f.free != nil {
+		f.free()
+	}
+}
+
+// GoAwayFrame is the representation of a [GOAWAY Frame]. The GOAWAY frame is
+// used to initiate shutdown of a connection or to signal serious error
+// conditions.
+//
+// [GOAWAY Frame]: https://httpwg.org/specs/rfc7540.html#GOAWAY
+type GoAwayFrame struct {
+	hdr          *FrameHeader
+	free         func()
+	LastStreamID uint32
+	Code         ErrCode
+	DebugData    []byte
+}
+
+func (f *GoAwayFrame) Header() *FrameHeader {
+	return f.hdr
+}
+
+func (f *GoAwayFrame) Free() {
+	if f.free != nil {
+		f.free()
+	}
+}
+
+// WindowUpdateFrame is the representation of a [WINDOW_UPDATE Frame]. The
+// WINDOW_UPDATE frame is used to implement flow control.
+//
+// [WINDOW_UPDATE Frame]: https://httpwg.org/specs/rfc7540.html#WINDOW_UPDATE
+type WindowUpdateFrame struct {
+	hdr *FrameHeader
+	Inc uint32
+}
+
+func (f *WindowUpdateFrame) Header() *FrameHeader {
+	return f.hdr
+}
+
+// ContinuationFrame is the representation of a [CONTINUATION Frame]. The
+// CONTINUATION frame is used to continue a sequence of header block fragments.
+//
+// [CONTINUATION Frame]: https://httpwg.org/specs/rfc7540.html#CONTINUATION
+type ContinuationFrame struct {
+	hdr      *FrameHeader
+	free     func()
+	HdrBlock []byte
+}
+
+func (f *ContinuationFrame) Header() *FrameHeader {
+	return f.hdr
+}
+
+func (f *ContinuationFrame) Free() {
+	if f.free != nil {
+		f.free()
+	}
+}
+
+// MetaHeadersFrame is the representation of one HEADERS frame and zero or more
+// contiguous CONTINUATION frames and the decoding of their HPACK-encoded
+// contents.  This frame type is not transmitted over the network and is only
+// generated by the ReadFrame() function.
+//
+// Since there is no underlying buffer in this Frame, Free() is a no-op.
+type MetaHeadersFrame struct {
+	hdr    *FrameHeader
+	Fields []hpack.HeaderField
+	// Truncated indicates whether the MetaHeadersFrame has been truncated due
+	// to being longer than the MaxHeaderListSize.
+	Truncated bool
+}
+
+func (f *MetaHeadersFrame) Header() *FrameHeader {
+	return f.hdr
+}
+
+func (f *MetaHeadersFrame) Free() {}
+
+// Framer encapsulates the functionality to read and write HTTP/2 frames.
+type Framer interface {
+	// ReadFrame returns grpchttp2.Frame. It is the caller's responsibility to
+	// call Frame.Free() once it is done using it. Note that once the mem
+	// package gets merged, this API will change in favor of Buffer.Free().
+	ReadFrame() (Frame, error)
+	// WriteData writes an HTTP/2 DATA frame to the stream.
+	// TODO: Once the mem package gets merged, data will change type to
+	// mem.BufferSlice.
+	WriteData(streamID uint32, endStream bool, data ...[]byte) error
+	// WriteData writes an HTTP/2 HEADERS frame to the stream.
+	// TODO: Once the mem package gets merged, headerBlock will change type to
```

Comment:
need to double checl, but TODOs without ldap (TODO(arvindbright)) or without a bug TODO(b/123456) will be tagged by the internal golinter during import

Reply:
That linter check might not block the import I feel. We have a lot of TODOs which don't have either. And adding bug numbers here will not make sense. So, we can either add username, or leave it as is.